src/bartiq/verification.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2024 PsiQuantum, Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+
+from qref.verification import verify_topology
+
+from . import Routine
+from .integrations.qref import bartiq_to_qref
+from .symbolics.backend import SymbolicBackend
+
+
+@dataclass
+class VerificationOutput:
+    """Dataclass containing the output of the verification."""
+
+    problems: list[str]
+
+    @property
+    def is_valid(self):
+        """If true, no issues has been found in the process of verification."""
+        return len(self.problems) == 0
+
+    def __bool__(self) -> bool:
+        return self.is_valid
+
+
+def verify_routine_topology(routine: Routine) -> VerificationOutput:
```

Comment:
**issue (nonblocking)**: Missing `Returns` section. For the sake of how the docs look, I would add a simple paragraph like:

```
Returns:
    An object indicating if the verification succeeded, containing a (possibly empty) list of found problems.
```